Add verbosity to smart-dispatch

bouthilx opened the pull request:
It is difficult to debug resuming while important process are taking place in the pbs script automatically built by SmartDispatch.

Adding:
- verbose to smart-dispatch/sd-launch-pbs command-line options
- debugging prints in epilog (based on verbosity option)

bouthilx committed:
Add verbosity to smart-dispatch

It is difficult to debug resuming while important process are taking place in
the pbs script automatically built by SmartDispatch.

We add verbose to smart-dispatch script and add debugging prints in epilog.

bouthilx commented:
@mgermain Small PR I took out of Slurm Backend PR

mgermain commented on scripts/smart-dispatch:
```diff
@@ -1,9 +1,10 @@
 #!/usr/bin/env python2
 # -*- coding: utf-8 -*-
 
+import argparse
```
Why change the order of the imports?

mgermain commented on scripts/sd-launch-pbs:
```diff
@@ -7,6 +7,10 @@ import logging
+
+logger = logging.getLogger()
+
+
```
remove the logging.root.setLevel(logging.INFO) below and make sure the default throughout the system is warning.

bouthilx replied:
@mgermain Done

mgermain replied:
Even in the rest of the smartdispatch to stay consistent?

bouthilx committed:
Remove forced logging level to INFO

The user can now specify which level of logging it wants with -vv
option. The default one is WARNING. If is it important for the user to
have INFO present in stderr, it can use -v or -vv.

mgermain commented on scripts/smart-dispatch:
```diff
@@ -29,19 +33,45 @@ TIMEOUT_EXIT_CODE = 124
 AUTORESUME_WORKER_CALL_PREFIX = 'timeout -s TERM {trigger_after} '.format(trigger_after=AUTORESUME_TRIGGER_AFTER)
 AUTORESUME_WORKER_CALL_SUFFIX = ' WORKER_PIDS+=" $!"'
-AUTORESUME_PROLOG = 'WORKER_PIDS=""' 
+AUTORESUME_PROLOG = """
```
Add the `\`  after the `"""` so there is no extra blank line later on.

`AUTORESUME_PROLOG = """\`

mgermain commented on scripts/smart-dispatch:
```diff
@@ -163,8 +190,12 @@ def main():
-        epilog = [AUTORESUME_EPILOG.format(launcher=LAUNCHER if args.launcher is None else args.launcher, path_job=path_job)]
+        prolog = [
+            AUTORESUME_PROLOG.format(verbose=str(args.verbose >= 2).lower())]
+        epilog = [
```
This project follows the entirety of PEP8 but the 80 columns.

mgermain commented:
After the tiny modifs mentioned this is ready to be merged.

bouthilx committed:
Revert wrapping lines longer than 79 chars.

I've been told the admins decided long ago not to follow rule E501. Even though
I disagree with this decision, I won't argue. :P

bouthilx committed:
Reformat VERBOSE to allow passing any level to sd-launch-pbs

VERBOSE in the PBS file was simply set to "true". Because of this we could only
pass args.verbose=0 or args.verbose=2 to sd-launch-pbs. VERBOSE is now set to
args.verbose and the command-line option {"", -v, -vv} is rebuilt in the PBS
script based on the value of args.parse.

bouthilx commented:
@mgermain I bit the bullet and reverted the E501 fixes. :stuck_out_tongue_winking_eye: I also changed the debugging logic in the epilog to support passing WARNING/INFO/DEBUG levels to sd-launch-pbs. Previously, it could only pass WARNING (no -v) or DEBUG (-vv).
